src/xenia/cpu/hir/instr.h

```diff
@@ -105,6 +104,17 @@ if both are constant, return nullptr, nullptr
   }
 
   Instr* GetDestDefSkipAssigns();
+  // returns [def op, constant]
+  std::pair<Value*, Value*> BinaryValueArrangeByDefOpAndConstant(
+      const OpcodeInfo* op_ptr) {
+    auto result = BinaryValueArrangeByDefiningOpcode(op_ptr);
+
+    if (!result.first) return result;
+    if (!result.second->IsConstant()) {
```

Comment:
Is there a chance that second can be null? What then?

Reply:
nope, it only returns NULL, NULL